PrimeNumbers.java

```diff
@@ -15,12 +15,41 @@ public static void myassert(boolean  x) {
     }
 
     private static int secondMaxSixDigitPrime(){
-        //WRITE YOUR CODE HERE
+        int num,i,check,max=0;
+        for (check = 999999; check >= 100000; check--){
+            int count = 0;
+            num = check/2;
+            for(i = 2; i <= num; i++){
+                if(check % i == 0){
+                    count++;
+                }
+            }
+            if(count == 0){
+                max ++;
+                if(max == 2){
+                    break;
+                }
+            }
+        }
+        return check;
     }
 
-
     private static List<Integer> fourDigitPrimes(){
-        //WRITE YOUR CODE HERE
+        int i,num,check;
+        ArrayList<Integer> primes = new ArrayList<>();
+        for (check = 1000; check <= 9999; check++){
+            int count = 0;
+            num = check/2;
+            for(i = 2; i <= num; i++){
```

Comment:
Can we reuse this code, the core prime # logic is there in both the functions. you can reuse the code by writing a separate function. Also you can break the loop even if one number is divisible, no need to check till n/2 and then add to primes list.